### PR TITLE
Tier and hotspot annotation added to P1 table

### PR DESCRIPTION
**Pull request by runjin326**
# Pull Request Template

## Description

This PR added a R script that added `Tier: xx` and `Cancer Hotspot` to the tables in P1:
`key_clinical_findings_output` and `all_findings_output`.

https://github.com/d3b-center/bixu-tracker/issues/1214

## Type of review for this PR
- [ ] Scientific / output review
- [ ] Code quality review

note: scientific/output review requires the lead scientists or analysts of the project to be more focus on the output of the PR where code quality review requires reviewer to make sure the PR is following the [contributing guideline](https://www.notion.so/d3b/contributing-721d172616b24ff395a64c18c58534f5) 

Code review part:
- In this PR, I tried to write the path as what was normally used in OMPARE - please double check whether all the paths look good.
- For `key_clinical_findings_output` and `all_findings_output`, I just use them as is since I didn't see those results writing out anywhere and they were being called in the `OMPARE.Rmd`.
- This code takes in `oncokb_consensus_annotated.txt` and the output is annotated `key_clinical_findings_output` and `all_findings_output`
- Two more files `hotspot_database_2017_indel.tsv` and `hotspot_database_2017_snv.tsv` were needed (from [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/e760c463dc1d638385d53e1778c8e4d600b21fa2/analyses/hotspots-detection/input/hotspots_database) ) - in the code, I put them in `data/hotspots_database`
- This code needs to run after OncoKB analysis. 

Scientific review part:
- For the Tier IV, in the ticket it says ~50% and ~100% - I used between 40%-60% and >90% as criteria. Please check whether that is ok.
- Also for Tier IV, it was listed that it should  not contain COSMIC ID - but actually since COSMIC does contain some germline variant as well (a good amount of entries also has dbSNP ID) - I currently took it out from the criteria. We can add it back also but I would recommend using a criteria that `contains COSM and not dbSNP`.
- I see that PolyPhen and SIFT was added for Tier IV. I currently have them in the critter but just want to confirm this is intended.
- Please check how the SNV and indel hotspots are being assigned and see whether that looks good.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

I tested with the `key_clinical_findings_output` downloaded from the patient 39 html. And for the `oncokb_consensus_annotated.txt`, I used a combined file with all `oncokb_consensus_annotated.txt` available in Cavatica (to make sure everything is captured correctly and wouldn't break).

- [ ] Unit test
- [ ] Integration test

**Test Configuration**:
* Environment:
* Test files:

**Task link/Screenshot/Terminal returns**:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have emojis in my PR and commits 😉

### CODE REVIEW

**Comment by komalsrathi** [comment]
> * For `key_clinical_findings_output` and `all_findings_output`, I just use them as is since I didn't see those results writing out anywhere and they were being called in the `OMPARE.Rmd`

Sounds good. I am not writing them out as they need to be called for every snv-caller type.

File: code/tier_classification/tier_classification.R
Comment (komalsrathi) [inline code/tier_classification/tier_classification.R:28]: Maybe we can filter the hotspots to genes corresponding to VUS/Mutations from all_findings_output (which is a superset of key_clinical_findings_output) so that the loop only runs on genes of interest. 

```suggestion
# subset to genes in all_findings_output corresponding to VUS and Mutation
genes_of_interest <- all_findings_output %>%
  filter(Type %in% c("VUS", "Mutation")) %>% 
  .$Aberration
hotspot_snv <- hotspot_snv %>% 
  filter(Hugo_Symbol %in% genes_of_interest)
hotspot_indel <- hotspot_indel %>%
  filter(Hugo_Symbol %in% hotspot_snv)
```

File: code/tier_classification/tier_classification.R
Comment (komalsrathi) [inline code/tier_classification/tier_classification.R:132]: and have a if statetment before all for-loops like this (because sometimes the filtering to genes of interest as suggested above could result in 0 rows): 

```suggestion
if(nrow(hotspot_snv)){
```

**Review by komalsrathi — APPROVED** [review]
Looks good. I can make the suggested changes on my end.

**Comment by komalsrathi** [comment]
@aadamk I will merge this after your approval.

**Comment by runjin326** [comment]
> Looks good. I can make the suggested changes on my end.
Thanks! :)

File: code/tier_classification/README.md
Comment (aadamk) [inline code/tier_classification/README.md:51]:
```suggestion
`hotspot_database_2017_indel.tsv` and `hotspot_database_2017_snv.tsv` derived from curated MSKCC data in OpenPBTA (https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/hotspots-detection) were used for annotating cancer hotspot mutations.
```

File: code/tier_classification/tier_classification.R
Comment (aadamk) [inline code/tier_classification/tier_classification.R:146]:
```suggestion
    dplyr::mutate(Variant_Properties= paste0("Cancer Hotspot Location; ", Variant_Properties))
```

File: code/tier_classification/tier_classification.R
Comment (aadamk) [inline code/tier_classification/tier_classification.R:153]:
```suggestion
    dplyr::mutate(Variant_Properties= paste0("Cancer Hotspot Location; ", Variant_Properties))
```

File: code/tier_classification/tier_classification.R
Comment (aadamk) [inline code/tier_classification/tier_classification.R:137]: I wonder if this could have been accomplished via a join between data tables in lieu of a for loop (e.g. matching on gene and position, then annotating variant_properties for genes that intersected between the two tables). Just a thought though not necessary to change if performance isn't an issue.

**Review by aadamk — APPROVED** [review]
Thanks @runjin326 and @komalsrathi.
I added a couple of minor suggestions and a comment (though no changes are necessary for the comment - may be useful if you or @komalsrathi decide to update the script further). 

1 key element that I changed came with the snv matching - I suggested `Cancer Hotspot Location` to replace `Cancer Hotspot`, as matching is occurring only at the level of genomic location, rather than on the precise variant type and amino acid change. 

1 update that we can make for hotspots (I'll open a separate ticket for this and it could be a new PR) is to use OpenPBTA's curated maf (https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/hotspots-detection/results/pbta-snv-scavenged-hotspots.maf.tsv.gz) to compare to the patient of interest MAF, matching on gene, location, HGVSp, and variant classification or type to unambiguously determine that the POI indeed harbors a given hotspot mutation.

File: code/tier_classification/tier_classification.R
Comment (aadamk) [inline code/tier_classification/tier_classification.R]:
```suggestion
    dplyr::mutate(Variant_Properties= paste0("Cancer Hotspot Location; ", Variant_Properties))
```

File: code/tier_classification/tier_classification.R
Comment (komalsrathi) [reply code/tier_classification/tier_classification.R:137]: I thought about that as well but because the `Details` column is a comma-separated string, it would be easier to use `grepl`

**Comment by runjin326** [comment]
@aadamk, thanks for reviewing this! Yes, I was a bit worried about just using the location and adding `location` is definitely better. I can work on unambiguously annotate hotspot implementing logics from OpenPBTA (or a more detailed database) to revolve that.

File: code/tier_classification/tier_classification.R
Comment (runjin326) [reply code/tier_classification/tier_classification.R:137]: Let me quickly update this!

**Pull request merged**
